### DISCUSSION TITLE
Connect to PulseAudio?

### DISCUSSION THREAD

**Discussion by rogerdahl** [diwic/dbus-rs#462] [Q&A] [answered]
Is it possible to connect to PulseAudio with dbus-rs?

I'm trying to connect to PulseAudio with a procedure similar to the one outlined here: https://www.freedesktop.org/wiki/Software/PulseAudio/Documentation/Developer/Clients/DBus/ConnectingToServer/

Following the procedure, I receive an address, like `unix:path=/run/user/1000/pulse/dbus-socket`. The Python example then shows a connection being created directly from this address, but I can't find a similar method in this crate?

**Comment by diwic** [comment]
I would try something like:
```
let mut channel = Channel::open_private('unix:path=/run/user/1000/pulse/dbus-socket')?;
channel.register()?;
Connection::from(channel)
```

**Reply by rogerdahl** [reply]
Thank you for the fast reply, and for dbus-rs. I gave your suggestion a try, and on the `channel.register()` call, I get an error:

```
D-Bus error: Method "Hello" with signature "" on interface "org.freedesktop.DBus" doesn't exist
```

Looking up the error, I found https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/issues/761, where someone says,

> Clients talk to PulseAudio on a peer-to-peer connection, not via a bus. The Hello method is relevant only for bus connections. 

Is this sort of peer-to-peer connection supported with dbus-rs?

**Reply by diwic** [reply, marked as answer]
Well, just skip the `register` call then:

```
let mut channel = Channel::open_private('unix:path=/run/user/1000/pulse/dbus-socket')?;
Connection::from(channel)
```

**Reply by rogerdahl** [reply]
Works beautifully. Thank you for the help!
